src/main/java/frc/robot/subsystems/climberSubsystem/ClimberSubsystem.java

```diff
@@ -0,0 +1,211 @@
+import frc.robot.utilities.DebugEntry;
+
+public class ClimberSubsystem extends SubsystemBase {
+  private PIDState pidState;
+  private CANSparkMax leftArmMotor;
+  private CANSparkMax rightArmMotor;
+  private ShuffleboardTab climberTab = Shuffleboard.getTab(this.getName());
+
+  private DebugEntry<Double> rightArmPoseEntry;
+  private DebugEntry<Double> leftArmPoseEntry;
+  private DebugEntry<Double> rightArmOutputEntry;
+  private DebugEntry<Double> leftArmOutputEntry;
+
+  public ClimberSubsystem() {
+    leftArmMotor = new CANSparkMax(ClimberConstants.LEFT_ARM_ID, MotorType.kBrushless);
+    rightArmMotor = new CANSparkMax(ClimberConstants.RIGHT_ARM_ID, MotorType.kBrushless);
+    configMotor(leftArmMotor, true);
+    configMotor(rightArmMotor, false);
+
+    rightArmPoseEntry =
+        new DebugEntry<Double>(
+            rightArmMotor.getEncoder().getPosition(), "right arm position", this);
+    leftArmPoseEntry =
+        new DebugEntry<Double>(rightArmMotor.getEncoder().getPosition(), "left arm position", this);
+    rightArmOutputEntry =
+        new DebugEntry<Double>(rightArmMotor.getAppliedOutput(), "right arm output", this);
+    leftArmOutputEntry =
+        new DebugEntry<Double>(rightArmMotor.getAppliedOutput(), "left arm output", this);
+  }
+
+  private void configMotor(CANSparkMax motor, boolean invert) {
+    motor.restoreFactoryDefaults();
+    motor.setIdleMode(IdleMode.kBrake);
+    SparkPIDController pidController = motor.getPIDController();
+    pidController.setP(ClimberConstants.CURRENT_PID[0]);
+    pidController.setI(ClimberConstants.CURRENT_PID[1]);
+    pidController.setD(ClimberConstants.CURRENT_PID[2]);
+    double armPosition;
+    if(invert){
+      armPosition = 1-motor.getAbsoluteEncoder().getPosition();
+    }else{
+      armPosition = motor.getAbsoluteEncoder().getPosition();
+    }
+
+    motor.setSoftLimit(SoftLimitDirection.kForward,(float) (Units.degreesToRotations(135)*ClimberConstants.GEAR_RATIO));
+    motor.setSoftLimit(SoftLimitDirection.kReverse, 0);
+
+    motor.getEncoder().setPosition(armPosition * ClimberConstants.GEAR_RATIO);
+    motor.setInverted(invert);
+    motor.setSmartCurrentLimit(40);
+  }
+
+  /**
+   * Positive is up, negative is down. Is even between arms.
+   *
+   * @param speed the speed to set (-1 to 1)
+   */
+  public void applyPercent(double speed) {
+    leftArmMotor.set(speed);
+    rightArmMotor.set(speed);
+  }
+
+  /**
+   * Positive is up, negative is down. Is even between arms.
+   *
+   * @param speed the speed to set (-1 to 1)
+   */
+  public void applyDiffertialPercent(DifferentialDemand demand) {
+    leftArmMotor.set(demand.left);
+    rightArmMotor.set(demand.right);
+  }
+
+  /**
+   * Positive is up, negative is down. Is even between arms.
+   *
+   * @param current the current to set
+   */
+  public void applyCurrentDemand(double current) {
+    leftArmMotor.getPIDController().setIAccum(0);
+    rightArmMotor.getPIDController().setIAccum(0);
```

Comment:
Since this is used with a run() Command instead of startEnd(), the I accumulation is cleared every robot period. Might not be what you want.

Reply:
Changed in latest commit